lib/ex_aws/s3/download.ex

```diff
@@ -0,0 +1,48 @@
+alias Experimental.GenStage
+
+defmodule ExAws.S3.Download do
+  @moduledoc false
+
+  defstruct [
+    :bucket,
+    :path,
+    :dest,
+    opts: [],
+  ]
+end
+
+defimpl ExAws.Operation, for: ExAws.S3.Download do
+
+  alias ExAws.S3.Download
+
+  def perform(op, config) do
+    {:ok, source} =
+      op.bucket
+      |> get_file_size(op.path, config)
+      |> Download.Source.start_link
+
+    {:ok, sink} = Download.Sink.start_link(op.dest)
+
+    for _ <- 1..Keyword.get(op.opts, :max_concurrency, 8) do
+      {:ok, worker} = Download.Worker.start_link(%{bucket: op.bucket, path: op.path, config: config})
```

Comment:
You can return `subscribe_to: [{source, min_demand: 0, max_demand: 1}]` as the third option in `init/1` so you don't need to subscribe with another request later.